src/main/kotlin/interpreter/pyobject/types/PyBytesType.kt

```diff
@@ -0,0 +1,45 @@
+package green.sailor.kython.interpreter.pyobject.types
+
+import green.sailor.kython.interpreter.Exceptions
+import green.sailor.kython.interpreter.functions.PyBuiltinFunction
+import green.sailor.kython.interpreter.iface.ArgType
+import green.sailor.kython.interpreter.iface.PyCallableSignature
+import green.sailor.kython.interpreter.pyobject.*
+import java.lang.IndexOutOfBoundsException
+
+object PyBytesType : PyType("bytes") {
+    override fun newInstance(kwargs: Map<String, PyObject>): PyObject {
+        val value = kwargs["value"] ?: error("Built-in signature mismatch")
+        when (value) {
```

Comment:
Don't use a when for a single conditional check. Instead, use a single if or a takeIf/Unless closure

Reply:
more checks will come when the other possible parameter types are added